core/benches/cluster_nodes.rs

```diff
@@ -0,0 +1,57 @@
+#![feature(test)]
+
+extern crate test;
+
+use {
+    rand::{seq::SliceRandom, Rng},
+    solana_core::{
+        cluster_nodes::{make_cluster, new_cluster_nodes, ClusterNodes},
+        retransmit_stage::RetransmitStage,
+    },
+    solana_gossip::contact_info::ContactInfo,
+    test::Bencher,
+};
+
+fn make_cluster_nodes<R: Rng>(rng: &mut R) -> (Vec<ContactInfo>, ClusterNodes<RetransmitStage>) {
+    let (nodes, stakes, cluster_info) = make_cluster(rng, 5_000);
+    let cluster_nodes = new_cluster_nodes::<RetransmitStage>(&cluster_info, &stakes);
+    (nodes, cluster_nodes)
+}
+
+#[bench]
+fn bench_get_retransmit_peers_deterministic(b: &mut Bencher) {
+    let mut rng = rand::thread_rng();
+    let (nodes, cluster_nodes) = make_cluster_nodes(&mut rng);
+
+    let mut shred_seed = [0u8; 32];
```

Comment:
Does it make sense to generate a different seed on every iteration to test different behaviors?

Reply:
I've changed this slightly to more accurately represent the seed generation for each case and to include multiple shreds in the calculation. I didn't want to include random generation of the inputs to `get_retransmit_peers_*` measurements.